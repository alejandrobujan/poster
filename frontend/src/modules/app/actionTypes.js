export const ERROR = 'poster/common/error';