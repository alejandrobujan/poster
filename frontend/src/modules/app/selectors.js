const getModuleState = state => state.app;

export const getError = state => getModuleState(state).error;

export const getFirstSearch = state => getModuleState(state).firstSearch;