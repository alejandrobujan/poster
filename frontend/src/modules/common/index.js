export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";