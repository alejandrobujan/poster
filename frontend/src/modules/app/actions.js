import * as actionTypes from './actionTypes';

export const error = error => ({
    type: actionTypes.ERROR,
    error
});

export const setFirstSearch = firstSearch => ({
    type: actionTypes.SET_FIRST_SEARCH,
    firstSearch
});