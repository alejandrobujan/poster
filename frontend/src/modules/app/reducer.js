import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    error: null,
    firstSearch: true
};

const error = (state = initialState.error, action) => {

    switch (action.type) {

        case actionTypes.ERROR:
            return action.error;

        default:
            return state;

    }

};    

const firstSearch = (state = initialState.firstSearch, action) => {

    switch (action.type) {

        case actionTypes.SET_FIRST_SEARCH:
            return action.firstSearch;

        default:
            return state;

    }

};    

const reducer = combineReducers({
    error,
    firstSearch
});

export default reducer;