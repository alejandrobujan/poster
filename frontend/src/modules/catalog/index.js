import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


