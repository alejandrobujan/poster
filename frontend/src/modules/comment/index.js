import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};