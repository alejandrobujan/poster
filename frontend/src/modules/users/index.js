import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Login} from './components/Login';
export {default as SignUp} from './components/SignUp';
export {default as Logout} from './components/Logout';
export {default as UpdateProfile} from './components/UpdateProfile';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};