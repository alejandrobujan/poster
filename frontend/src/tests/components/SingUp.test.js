import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';
import renderer from "react-test-renderer";
import store from "../../store";
import { SignUp } from "../../modules/users";
import { App } from "../../modules/app";

import {createMemoryHistory} from 'history';

describe("SignUp", () => {
	it("renders correctly", () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<MemoryRouter>
						<SignUp/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

    it("do sign up", async () => {
        const history = createMemoryHistory();

		render(
			<Provider store={store}>
				<MemoryRouter>
					<App/>
				</MemoryRouter>
			</Provider>
		);

		const loginLink = screen.getByText('Login');

		fireEvent.click(loginLink);
 
        await waitFor(() => {
            const signUp = screen.getByText(/Sign up/);
			fireEvent.click(signUp);
        }).then(async () => {
			const login = screen.getByLabelText('Login');

        fireEvent.change(login, {
			target: { value: "Pepe" },
		});

        const email = screen.getByLabelText('Email');

        fireEvent.change(email, {
			target: { value: "Pepe@pepe.com" },
		});

        const password = screen.getByLabelText('Password');

        fireEvent.change(password, {
			target: { value: "Pepe" },
		});

        const confirmPassword = screen.getByLabelText('Confirm password');

        fireEvent.change(confirmPassword, {
			target: { value: "Pepe" },
		});

        const firstName = screen.getByLabelText('First name');

        fireEvent.change(firstName, {
			target: { value: "Pepe" },
		});

        const lastName = screen.getByLabelText('Last name');

        fireEvent.change(lastName, {
			target: { value: "Pepe" },
		});

        const submit = screen.getByRole('button', {name: 'Submit'});

        fireEvent.click(submit);

	
		})/*.then(async () => {
			await waitFor(() => {
				screen.getByText(/No posts found/i);
			});
		});

        /*const login = screen.getByLabelText('Login');

        fireEvent.change(login, {
			target: { value: "Pepe" },
		});

        const email = screen.getByLabelText('Email');

        fireEvent.change(email, {
			target: { value: "Pepe@pepe.com" },
		});

        const password = screen.getByLabelText('Password');

        fireEvent.change(password, {
			target: { value: "Pepe" },
		});

        const confirmPassword = screen.getByLabelText('Confirm password');

        fireEvent.change(confirmPassword, {
			target: { value: "Pepe" },
		});

        const firstName = screen.getByLabelText('First name');

        fireEvent.change(firstName, {
			target: { value: "Pepe" },
		});

        const lastName = screen.getByLabelText('Last name');

        fireEvent.change(lastName, {
			target: { value: "Pepe" },
		});

        const submit = screen.getByRole('button', {name: 'Submit'});

        fireEvent.click(submit);

		await waitFor(() => {
			expect(history.location.pathname).toBe('/poster');
		});

        //expect(history.location.pathname).toEqual('/poster');*/



	});

    
    /*it("do sign up wrong", async () => {
        const history = createMemoryHistory();

		render(
			<Provider store={store}>
				<MemoryRouter>
					<SignUp/>
				</MemoryRouter>
			</Provider>
		);

        const login = screen.getByLabelText('Login');

        fireEvent.change(login, {
			target: { value: "Pepe" },
		});

        const email = screen.getByLabelText('Email');

        fireEvent.change(email, {
			target: { value: "Pepe" },
		});

        const password = screen.getByLabelText('Password');

        fireEvent.change(password, {
			target: { value: "Pepe" },
		});

        const confirmPassword = screen.getByLabelText('Confirm password');

        fireEvent.change(confirmPassword, {
			target: { value: "Pepe" },
		});

        const firstName = screen.getByLabelText('First name');

        fireEvent.change(firstName, {
			target: { value: "Pepe" },
		});

        const lastName = screen.getByLabelText('Last name');

        fireEvent.change(lastName, {
			target: { value: "Pepe" },
		});

        const submit = screen.getByRole('button', {name: 'Submit'});

        fireEvent.click(submit);

        expect(history.location.pathname).toEqual('/poster/users/signup');

        const alert = await screen.findByRole('alert');

	});*/
});