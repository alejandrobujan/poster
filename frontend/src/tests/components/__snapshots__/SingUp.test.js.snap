// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="card bg-light border-dark m-4"
  >
    <h5
      className="card-header text-center"
    >
      Sign up
    </h5>
    <div
      className="card-body"
    >
      <form
        className="needs-validation"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="login"
          >
            Login
          </label>
          <div
            className="col-md-6"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="login"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The login is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="email"
          >
            Email
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="email"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The email is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="password"
          >
            Password
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The password is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="confirmPassword"
          >
            Confirm password
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="confirmPassword"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The password confirmation is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="firstName"
          >
            First name
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="firstName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The first name is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="lastName"
          >
            Last name
          </label>
          <div
            className="col-md-6"
          >
            <input
              className="form-control"
              id="lastName"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
            <div
              className="invalid-feedback"
            >
              The last name is required
            </div>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-md-6 col-form-label"
            htmlFor="avatar"
          >
            Avatar
          </label>
          <div
            className="col-md-6"
          >
            <input
              accept="image/*"
              id="avatar"
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
